### PR TITLE
Add Opentelemetry Integration docs

### LINKED ISSUE #641: [Enhancement] OpenTelemetry support
### Description

Document OpenTelemetry in 1.8.0

### Slack discussion

_No response_

### PR DESCRIPTION
**Pull request by 4molybdenum2**
Signed-off-by: Tathagata Paul <tathagatapaul7@gmail.com>

## Related issue #
 Related to PR [#3910](https://github.com/kyverno/kyverno/pull/3910)
Closes #641 

## Proposed Changes
- [x] Remove metrics refresh interval because it will no longer be supported
- [x] Add documentation for integration of Opentelemetry into Kyverno

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/kyverno/website/blob/main/CONTRIBUTING.md).
- [x] I have inspected the website preview for accuracy.
- [x] I have signed off my issue.

### CODE REVIEW

**Comment by chipzoller** [comment]
This cannot be merged until the referenced PR is merged.

**Comment by realshuting** [comment]
@4molybdenum2 - can you please convert the draft to the PR if it's ready for review?

**Comment by 4molybdenum2** [comment]
done

File: content/en/docs/Monitoring/opentelemetry-integration.md [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+### Setting up Jaeger
+
+The traces are pushed to the Jaeger backend on port **14250**. To install Jaeger:
+
+- First install the Jaeger Operator
+
+```shell
+kubectl create namespace observability
+kubectl create -n observability -f https://github.com/jaegertracing/jaeger-operator/releases/download/v1.33.0/jaeger-operator.yaml
```
Comment (realshuting) [inline content/en/docs/Monitoring/opentelemetry-integration.md:252]: Got the following error when running this command:
```
➜  ~ kubectl create -n observability -f https://github.com/jaegertracing/jaeger-operator/releases/download/v1.33.0/jaeger-operator.yaml
customresourcedefinition.apiextensions.k8s.io/jaegers.jaegertracing.io created
serviceaccount/jaeger-operator created
role.rbac.authorization.k8s.io/leader-election-role created
role.rbac.authorization.k8s.io/prometheus created
clusterrole.rbac.authorization.k8s.io/jaeger-operator-metrics-reader created
clusterrole.rbac.authorization.k8s.io/manager-role created
clusterrole.rbac.authorization.k8s.io/proxy-role created
rolebinding.rbac.authorization.k8s.io/leader-election-rolebinding created
rolebinding.rbac.authorization.k8s.io/prometheus created
clusterrolebinding.rbac.authorization.k8s.io/jaeger-operator-proxy-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/manager-rolebinding created
service/jaeger-operator-metrics created
service/jaeger-operator-webhook-service created
deployment.apps/jaeger-operator created
mutatingwebhookconfiguration.admissionregistration.k8s.io/jaeger-operator-mutating-webhook-configuration created
validatingwebhookconfiguration.admissionregistration.k8s.io/jaeger-operator-validating-webhook-configuration created
unable to recognize "https://github.com/jaegertracing/jaeger-operator/releases/download/v1.33.0/jaeger-operator.yaml": no matches for kind "Certificate" in version "cert-manager.io/v1"
unable to recognize "https://github.com/jaegertracing/jaeger-operator/releases/download/v1.33.0/jaeger-operator.yaml": no matches for kind "Issuer" in version "cert-manager.io/v1"
```

Comment (realshuting) [inline content/en/docs/Monitoring/opentelemetry-integration.md:252]: And the deployment gave me the following error:
```
Events:
  Type     Reason       Age                  From               Message
  ----     ------       ----                 ----               -------
  Normal   Scheduled    2m51s                default-scheduler  Successfully assigned observability/jaeger-operator-db884b6b6-qmb77 to k3d-k3s-default-agent-1
  Warning  FailedMount  48s                  kubelet            Unable to attach or mount volumes: unmounted volumes=[cert], unattached volumes=[cert kube-api-access-s5lb9]: timed out waiting for the condition
  Warning  FailedMount  43s (x9 over 2m51s)  kubelet            MountVolume.SetUp failed for volume "cert" : secret "jaeger-operator-service-cert" not found
```

**Review by realshuting — COMMENTED** [review]
@4molybdenum2 - I went through the setup and faced a few issues, see below.

File: content/en/docs/Monitoring/opentelemetry-integration.md [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+  type: ClusterIP
+```
+
+This defined a service for the discovery of the collector deployment.
+
+
+### Setting up Kyverno and passing required flags
+
+In the ```install.yaml``` for Kyverno. 
+- Pass the flag ```metricsPort``` to defined the Opentelemetry Collector endpoint for collecting metrics.
```
[the remark this answers is not in the document]
Comment (4molybdenum2) [reply content/en/docs/Monitoring/opentelemetry-integration.md:129]: default is port 8000. I will mention this in docs

File: content/en/docs/Monitoring/opentelemetry-integration.md (hunk 1 above)
Comment (4molybdenum2) [reply content/en/docs/Monitoring/opentelemetry-integration.md:252]: Cert manager is needed for the installation. I will add this to the docs

**Comment by 4molybdenum2** [comment]
@realshuting made a few changes...one the issues that I could understand was occuring was because cert-manager was not installed. Can you try to install that and check if it is working?

File: content/en/docs/Monitoring/opentelemetry-integration.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
+---
+title: Opentelemetry Integration
+description: Kyverno is instrumented with the Opentelemetry SDK. This helps us currently in exposing metrics and traces to an Opentelemetry Collector. This guide will show you how to set up a simple Opentelemetry Collector as a deployment and export Kyverno Metrics and Traces to it. This is just an example and must not be used in production.
+weight: 65
```
Comment (chipzoller) [inline content/en/docs/Monitoring/opentelemetry-integration.md:4]: Change weight to 61.

File: content/en/docs/Monitoring/opentelemetry-integration.md [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+---
+
+## Opentelemetry Setup
+
+We will work with a few yaml files are configurations to setup our collector in the kyverno namespace. The required configurations are listed below.
+
+### Install Cert-Manager
+
+```
```
Comment (chipzoller) [inline content/en/docs/Monitoring/opentelemetry-integration.md:13]: Code blocks need syntax. Use `sh` for this one.

File: content/en/docs/Monitoring/opentelemetry-integration.md [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.9.1/cert-manager.yaml
+```
+
+### Config file for Opentelemetry Collector
+
+Create a ```configmap.yaml``` with the following content:
```
Comment (chipzoller) [inline content/en/docs/Monitoring/opentelemetry-integration.md:19]: In-line code formatting such as a file name should only be wrapped in single back ticks, not three. Use of three is for a code block.

Comment (chipzoller) [reply content/en/docs/Monitoring/opentelemetry-integration.md:19]: New lines should separate all code blocks.

**Review by chipzoller — CHANGES_REQUESTED** [review]

**Comment by realshuting** [comment]
@4molybdenum2 - do we need to update the grafana dashboard JSON file https://github.com/kyverno/grafana-dashboard/blob/master/grafana/dashboard.json?

**Comment by 4molybdenum2** [comment]
@realshuting I have not dealt with the grafana files while working on the refactoring, so I am not entirely sure if anything is required.

**Comment by chipzoller** [comment]
@4molybdenum2 please help close this PR out as it is needed to release 1.8.0 docs.

**Comment by 4molybdenum2** [comment]
@chipzoller I couldn't get to clear this as I am swamped by my exams 🥲. I will sort this out by day after tomorrow

**Comment by 4molybdenum2** [comment]
@chipzoller made changes to the docs according to the review

File: content/en/docs/Monitoring/_index.md [hunk 6]
```diff
@@ -99,17 +99,4 @@ config:
...
-...
-config:
-  # rate at which metrics should reset so as to clean up the memory footprint of kyverno metrics, if you might be expecting high memory footprint of Kyverno's metrics.
-  metricsRefreshInterval: 24h 
-  #Default: 0, no refresh of metrics
-...
-```
-
-> You still would not lose your previous metrics as your metrics get persisted in the Prometheus backend.
-
```
Comment (chipzoller) [inline content/en/docs/Monitoring/_index.md:114]: @4molybdenum2 I'm sorry these comments are coming so late, but can you explain why this section of the docs should be removed?

**Review by chipzoller — COMMENTED** [review]
Have some questions if you please.

File: content/en/docs/Monitoring/_index.md (hunk 6 above)
Comment (4molybdenum2) [reply content/en/docs/Monitoring/_index.md:114]: @chipzoller this parameter is deprecated currently, we don't need to configure this anymore

Comment (chipzoller) [reply content/en/docs/Monitoring/_index.md:114]: Deprecated or removed?

**Pull request closed without merging**